src/module.c

```diff
@@ -10760,6 +10872,16 @@ int moduleLoad(const char *path, void **module_argv, int module_argc) {
         serverLog(LL_WARNING, "Module %s failed to load: %s", path, dlerror());
         return C_ERR;
     }
+    /* If this is a loadex command we want to populate server.module_configs_queue with sds NAME VALUE pairs.
+     * We also want to increment argv to just after ARGS. */
+    if (is_loadex) {
+        /* This call will also increment the module_argv pointer to just after ARGS. */
+        if (parseLoadexArguments((RedisModuleString ***) &module_argv, &module_argc)) {
```

Comment:
I still believe this logic can be in the moduleCommand() function. It's then possible to have the parseLoadexArguments actually reply with a useful human message. Generally the best practice is to do validation of input parameters before any actual work.

(I think on failure this leaks the open handle too)